--- a/ci.eff
+++ b/ci.eff
@@ -1,6 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
+
+<!-- CI effect
+  The CI uses a custom shader to render the radar screen.
+  This is FG 2020.3.13 Effects/model-default.eff with minimal changes
+  to include this custom code.
+-->
+
 <PropertyList>
 	<name>Effects/model-default</name>
 	<!-- 	<inherits-from>Effects/shadow</inherits-from> -->
 	<parameters>
 		<texture n ="0">
@@ -55,11 +62,20 @@
 		<geo_light_radius type="float">0.0</geo_light_radius>
 		<geo_ambience type="float">0.0</geo_ambience>
 		<quality_level><use>/sim/rendering/shaders/landmass</use></quality_level>
 		<tquality_level><use>/sim/rendering/shaders/transition</use></tquality_level>
 		<!-- 		END fog include -->
-	</parameters>
+
+    <!-- CI properties -->
+    <polaroid-filter><use>/instrumentation/radar/polaroid-filter</use></polaroid-filter>
+    <display-mode><use>/instrumentation/radar/effect/mode</use></display-mode>
+    <time1><use>/instrumentation/radar/effect/time1</use></time1>
+    <time2><use>/instrumentation/radar/effect/time2</use></time2>
+    <beam-pos><use>/instrumentation/radar/effect/beam-pos-norm</use></beam-pos>
+    <beam-dir><use>/instrumentation/radar/effect/beam-dir</use></beam-dir>
+    <!-- end CI properties -->
+  </parameters>
 
   <technique n="5">
     <predicate>
       <and>
         <property>/sim/rendering/shaders/skydome</property>
@@ -119,16 +135,20 @@
       </texture-unit>
       <vertex-program-two-side>
         <use>vertex-program-two-side</use>
       </vertex-program-two-side>
       <program>
-        <vertex-shader>Shaders/generic-ALS-base.vert</vertex-shader>
-        <fragment-shader>Shaders/model-ALS-base.frag</fragment-shader>
+        <!-- CI shaders: add custom shaders (ci.{vert,frag}),
+             use modified default shaders which call the CI code -->
+        <vertex-shader>Aircraft/JA37/Models/Effects/CI/generic-ALS-base.vert</vertex-shader>
+        <vertex-shader>Aircraft/JA37/Models/Effects/CI/ci.vert</vertex-shader>
+        <fragment-shader>Aircraft/JA37/Models/Effects/CI/model-ALS-base.frag</fragment-shader>
         <fragment-shader>Shaders/hazes.frag</fragment-shader>
         <fragment-shader>Shaders/secondary_lights.frag</fragment-shader>
         <fragment-shader>Shaders/noise.frag</fragment-shader>
         <fragment-shader>Shaders/filters-ALS.frag</fragment-shader>
+        <fragment-shader>Aircraft/JA37/Models/Effects/CI/ci.frag</fragment-shader>
       </program>
       <uniform>
         <name>visibility</name>
         <type>float</type>
         <value><use>visibility</use></value>
@@ -351,90 +371,67 @@
       <uniform>
         <name>tquality_level</name>
         <type>int</type>
         <value><use>tquality_level</use></value>
       </uniform>
+
+      <!-- CI uniforms -->
+      <uniform>
+        <name>display_mode</name>
+        <type>int</type>
+        <value><use>display-mode</use></value>
+      </uniform>
+      <uniform>
+        <name>current_time1</name>
+        <type>float</type>
+        <value><use>time1</use></value>
+      </uniform>
+      <uniform>
+        <name>current_time2</name>
+        <type>float</type>
+        <value><use>time2</use></value>
+      </uniform>
+      <uniform>
+        <name>polaroid_filter</name>
+        <type>float</type>
+        <value><use>polaroid-filter</use></value>
+      </uniform>
+      <uniform>
+        <name>beam_pos</name>
+        <type>float</type>
+        <value><use>beam-pos</use></value>
+      </uniform>
+      <uniform>
+        <name>beam_dir</name>
+        <type>int</type>
+        <value><use>beam-dir</use></value>
+      </uniform>
+      <!-- end CI uniforms -->
     </pass>
   </technique>
 
-<technique n="10">
-	<predicate>
-		<and>
-			<property>/sim/rendering/rembrandt/enabled</property>
-			<or>
-				<less-equal>
-					<value type="float">2.0</value>
-					<glversion/>
-				</less-equal>
-				<and>
-					<extension-supported>GL_ARB_shader_objects</extension-supported>
-					<extension-supported>GL_ARB_shading_language_100</extension-supported>
-					<extension-supported>GL_ARB_vertex_shader</extension-supported>
-					<extension-supported>GL_ARB_fragment_shader</extension-supported>
-				</and>
-			</or>
-		</and>
-	</predicate>
-	<pass>
-		<lighting>false</lighting>
-		<material>
-			<ambient><use>material/ambient</use></ambient>
-			<diffuse><use>material/diffuse</use></diffuse>
-			<specular><use>material/specular</use></specular>
-			<emissive><use>material/emissive</use></emissive>
-			<shininess><use>material/shininess</use></shininess>
-			<color-mode>ambient-and-diffuse</color-mode>
-		</material>
-		<blend>false</blend>
-		<alpha-test>true</alpha-test>
-		<shade-model>smooth</shade-model>
-		<cull-face><use>cull-face</use></cull-face>
-		<render-bin>
-			<bin-number>1</bin-number>
-			<bin-name>RenderBin</bin-name>
-		</render-bin>
-		<texture-unit>
-			<unit>0</unit>
-			<type><use>texture[0]/type</use></type>
-			<image><use>texture[0]/image</use></image>
-			<filter><use>texture[0]/filter</use></filter>
-			<wrap-s><use>texture[0]/wrap-s</use></wrap-s>
-			<wrap-t><use>texture[0]/wrap-t</use></wrap-t>
-			<internal-format>
-				<use>texture[0]/internal-format</use>
-			</internal-format>
-			<environment>
-				<mode>modulate</mode>
-			</environment>
-		</texture-unit>
-		<vertex-program-two-side>
-			<use>vertex-program-two-side</use>
-		</vertex-program-two-side>
-		<program n="0">
-			<vertex-shader n="0">Shaders/deferred-gbuffer.vert</vertex-shader>
-			<fragment-shader>Shaders/deferred-gbuffer.frag</fragment-shader>
-			<fragment-shader>Shaders/gbuffer-functions.frag</fragment-shader>
-			<fragment-shader>Shaders/gbuffer-encode.frag</fragment-shader>
-		</program>
-		<uniform>
-			<name>texture</name>
-			<type>sampler-2d</type>
-			<value type="int">0</value>
-		</uniform>
-		<!-- uniform>
-			<name>materialID</name>
-			<type>int</type>
-			<value><use>material-id</use></value>
-		</uniform -->
-	</pass>
-</technique>
+  <!-- Rembrandt removed here, Viggen is not even close to rembrandt compatible. Sorry. -->
 
 	<technique n="11">
 		<predicate>
 			<and>
-				<property>/sim/rendering/shaders/quality-level</property>
-				<property>/sim/rendering/shaders/generic</property>
+				<or>
+					<and>
+						<equal>
+							<property>/ja37/supported/compositor</property>
+							<value type="bool">false</value>
+						</equal>
+						<property>/sim/rendering/shaders/quality-level</property>
+						<property>/sim/rendering/shaders/model</property>
+					</and>
+					<!-- FG 2020.4+ compatibility - this is not a compositor shader,
+						but this at least loads the shader correctly on 2020.4 -->
+					<and>
+						<property>/ja37/supported/compositor</property>
+						<property>/sim/rendering/shaders/use-shaders</property>
+					</and>
+				</or>
 				<or>
 					<less-equal>
 						<value type="float">2.0</value>
 						<glversion/>
 					</less-equal>
@@ -521,14 +518,18 @@
 			</texture-unit>
 			<vertex-program-two-side>
 				<use>vertex-program-two-side</use>
 			</vertex-program-two-side>
 			<program n="0">
-<!-- 				<vertex-shader n="0">Shaders/include_fog.vert</vertex-shader> -->			<!--fog include-->
-				<vertex-shader n="1">Shaders/default.vert</vertex-shader>
-				<fragment-shader n="0">Shaders/include_fog.frag</fragment-shader>				<!--fog include-->
-				<fragment-shader n="1">Shaders/default.frag</fragment-shader>
+        <!-- CI shaders: add custom shaders (ci.{vert,frag}),
+             use modified default shaders which call the CI code -->
+        <vertex-shader n="1">Aircraft/JA37/Models/Effects/CI/default.vert</vertex-shader>
+        <vertex-shader n="2">Aircraft/JA37/Models/Effects/CI/ci.vert</vertex-shader>
+        <fragment-shader n="0">Shaders/include_fog.frag</fragment-shader>				<!--fog include-->
+        <fragment-shader n="1">Aircraft/JA37/Models/Effects/CI/default.frag</fragment-shader>
+        <fragment-shader n="2">Aircraft/JA37/Models/Effects/CI/ci.frag</fragment-shader>
+        <fragment-shader>Shaders/noise.frag</fragment-shader>
 			</program>
 			<!-- 			BEGIN fog include -->
 			<uniform>
 				<name>visibility</name>
 				<type>float</type>
@@ -582,10 +583,43 @@
 				<type>int</type>
 				<value>
 					<use>material/color-mode-uniform</use>
 				</value>
 			</uniform>
+
+      <!-- CI uniforms -->
+      <uniform>
+        <name>display_mode</name>
+        <type>int</type>
+        <value><use>display-mode</use></value>
+      </uniform>
+      <uniform>
+        <name>current_time1</name>
+        <type>float</type>
+        <value><use>time1</use></value>
+      </uniform>
+      <uniform>
+        <name>current_time2</name>
+        <type>float</type>
+        <value><use>time2</use></value>
+      </uniform>
+      <uniform>
+        <name>polaroid_filter</name>
+        <type>float</type>
+        <value><use>polaroid-filter</use></value>
+      </uniform>
+      <uniform>
+        <name>beam_pos</name>
+        <type>float</type>
+        <value><use>beam-pos</use></value>
+      </uniform>
+      <uniform>
+        <name>beam_dir</name>
+        <type>int</type>
+        <value><use>beam-dir</use></value>
+      </uniform>
+      <!-- end CI uniforms -->
 		</pass>
 	</technique>
 	<technique n="13">
 		<pass>
 			<lighting>true</lighting>
