--- a/ci.eff
+++ b/ci.eff
@@ -1,8 +1,15 @@
 <?xml version="1.0" encoding="utf-8"?>
+
+<!-- CI effect
+  The CI uses a custom shader to render the radar screen.
+  This is FG 2020.4 Effects/model-default.eff with minimal changes
+  to include this custom code.
+-->
+
 <PropertyList>
-	<name>Effects/model-default</name>
+	<name>Aircraft/JA37/Models/Effects/CI</name>
 	<!-- 	<inherits-from>Effects/shadow</inherits-from> -->
 	<parameters>
 		<texture n ="0">
 			<type>white</type>
 		</texture>
@@ -63,11 +70,20 @@
         </shadows_enabled>
         <sun_atlas_size>
           <use>/sim/rendering/shadows/sun-atlas-size</use>
         </sun_atlas_size>
         <!-- END shadows include -->
-	</parameters>
+
+    <!-- CI properties -->
+    <polaroid-filter><use>/instrumentation/radar/polaroid-filter</use></polaroid-filter>
+    <display-mode><use>/instrumentation/radar/effect/mode</use></display-mode>
+    <time1><use>/instrumentation/radar/effect/time1</use></time1>
+    <time2><use>/instrumentation/radar/effect/time2</use></time2>
+    <beam-pos><use>/instrumentation/radar/effect/beam-pos-norm</use></beam-pos>
+    <beam-dir><use>/instrumentation/radar/effect/beam-dir</use></beam-dir>
+    <!-- end CI properties -->
+  </parameters>
 
   <technique n="5">
     <predicate>
       <and>
         <property>/sim/rendering/shaders/skydome</property>
@@ -127,19 +143,23 @@
       </texture-unit>
       <vertex-program-two-side>
         <use>vertex-program-two-side</use>
       </vertex-program-two-side>
       <program>
-        <vertex-shader>Shaders/generic-ALS-base.vert</vertex-shader>
+        <!-- CI shaders: add custom shaders (ci.{vert,frag}),
+             use modified default shaders which call the CI code -->
+        <vertex-shader>Aircraft/JA37/Models/Effects/CI/generic-ALS-base.vert</vertex-shader>
         <vertex-shader>Shaders/shadows-include.vert</vertex-shader>
-        <fragment-shader>Shaders/model-ALS-base.frag</fragment-shader>
+        <vertex-shader>Aircraft/JA37/Models/Effects/CI/ci.vert</vertex-shader>
+        <fragment-shader>Aircraft/JA37/Models/Effects/CI/model-ALS-base.frag</fragment-shader>
         <fragment-shader>Shaders/hazes.frag</fragment-shader>
         <fragment-shader>Shaders/secondary_lights.frag</fragment-shader>
         <fragment-shader>Shaders/noise.frag</fragment-shader>
         <fragment-shader>Shaders/filters-ALS.frag</fragment-shader>
         <fragment-shader>Shaders/shadows-include.frag</fragment-shader>
         <fragment-shader>Shaders/clustered-include.frag</fragment-shader>
+        <fragment-shader>Aircraft/JA37/Models/Effects/CI/ci.frag</fragment-shader>
       </program>
       <uniform>
         <name>visibility</name>
         <type>float</type>
         <value><use>visibility</use></value>
@@ -383,17 +403,54 @@
         <value>
           <use>sun_atlas_size</use>
         </value>
       </uniform>
       <!-- END shadows include -->
+
+      <!-- CI uniforms -->
+      <uniform>
+        <name>display_mode</name>
+        <type>int</type>
+        <value><use>display-mode</use></value>
+      </uniform>
+      <uniform>
+        <name>current_time1</name>
+        <type>float</type>
+        <value><use>time1</use></value>
+      </uniform>
+      <uniform>
+        <name>current_time2</name>
+        <type>float</type>
+        <value><use>time2</use></value>
+      </uniform>
+      <uniform>
+        <name>polaroid_filter</name>
+        <type>float</type>
+        <value><use>polaroid-filter</use></value>
+      </uniform>
+      <uniform>
+        <name>beam_pos</name>
+        <type>float</type>
+        <value><use>beam-pos</use></value>
+      </uniform>
+      <uniform>
+        <name>beam_dir</name>
+        <type>int</type>
+        <value><use>beam-dir</use></value>
+      </uniform>
+      <!-- end CI uniforms -->
     </pass>
   </technique>
 
 	<technique n="11">
 		<predicate>
 			<and>
-				<property>/sim/rendering/shaders/use-shaders</property>
+				<or>
+					<!-- if this property is set (default=yes), always keep this shader on -->
+					<property>/ja37/displays/use-CI-shader-on-min-settings</property>
+					<property>/sim/rendering/shaders/use-shaders</property>
+				</or>
 				<or>
 					<less-equal>
 						<value type="float">2.0</value>
 						<glversion/>
 					</less-equal>
@@ -480,17 +537,22 @@
 			</texture-unit>
 			<vertex-program-two-side>
 				<use>vertex-program-two-side</use>
 			</vertex-program-two-side>
 			<program n="0">
+				<!-- CI shaders: add custom shaders (ci.{vert,frag}),
+					use modified default shaders which call the CI code -->
 <!-- 				<vertex-shader n="0">Shaders/include_fog.vert</vertex-shader> -->			<!--fog include-->
-				<vertex-shader n="1">Shaders/default.vert</vertex-shader>
+				<vertex-shader n="1">Aircraft/JA37/Models/Effects/CI/default.vert</vertex-shader>
 				<vertex-shader n="2">Shaders/shadows-include.vert</vertex-shader>
+				<vertex-shader n="3">Aircraft/JA37/Models/Effects/CI/ci.vert</vertex-shader>
 				<fragment-shader n="0">Shaders/include_fog.frag</fragment-shader>				<!--fog include-->
-				<fragment-shader n="1">Shaders/default.frag</fragment-shader>
+				<fragment-shader n="1">Aircraft/JA37/Models/Effects/CI/default.frag</fragment-shader>
 				<fragment-shader n="2">Shaders/shadows-include.frag</fragment-shader>
 				<fragment-shader n="3">Shaders/clustered-include.frag</fragment-shader>
+				<fragment-shader n="4">Aircraft/JA37/Models/Effects/CI/ci.frag</fragment-shader>
+				<fragment-shader n="5">Shaders/noise.frag</fragment-shader>
 			</program>
 			<!-- 			BEGIN fog include -->
 			<uniform>
 				<name>visibility</name>
 				<type>float</type>
@@ -565,10 +627,43 @@
 			  <value>
 				<use>sun_atlas_size</use>
 			  </value>
 			</uniform>
 			<!-- END shadows include -->
+
+      <!-- CI uniforms -->
+      <uniform>
+        <name>display_mode</name>
+        <type>int</type>
+        <value><use>display-mode</use></value>
+      </uniform>
+      <uniform>
+        <name>current_time1</name>
+        <type>float</type>
+        <value><use>time1</use></value>
+      </uniform>
+      <uniform>
+        <name>current_time2</name>
+        <type>float</type>
+        <value><use>time2</use></value>
+      </uniform>
+      <uniform>
+        <name>polaroid_filter</name>
+        <type>float</type>
+        <value><use>polaroid-filter</use></value>
+      </uniform>
+      <uniform>
+        <name>beam_pos</name>
+        <type>float</type>
+        <value><use>beam-pos</use></value>
+      </uniform>
+      <uniform>
+        <name>beam_dir</name>
+        <type>int</type>
+        <value><use>beam-dir</use></value>
+      </uniform>
+      <!-- end CI uniforms -->
 		</pass>
 	</technique>
 	<technique n="13">
 		<pass>
 			<lighting>true</lighting>
@@ -649,71 +744,7 @@
            effect?
         -->
 		</pass>
 	</technique>
 
-    <technique n="109">
-      <scheme>hdr-geometry</scheme>
-      <pass>
-        <!-- Reverse floating point depth buffer -->
-        <depth>
-          <function>gequal</function>
-          <near>1.0</near>
-          <far>0.0</far>
-        </depth>
-        <stencil>
-          <function>always</function>
-          <value>8</value>
-          <pass>replace</pass>
-        </stencil>
-        <texture-unit>
-          <unit>0</unit>
-          <type><use>texture[0]/type</use></type>
-          <image><use>texture[0]/image</use></image>
-          <filter><use>texture[0]/filter</use></filter>
-          <wrap-s><use>texture[0]/wrap-s</use></wrap-s>
-          <wrap-t><use>texture[0]/wrap-t</use></wrap-t>
-        </texture-unit>
-        <blend>0</blend>
-        <rendering-hint>opaque</rendering-hint>
-        <cull-face><use>cull-face</use></cull-face>
-        <program>
-          <vertex-shader>Shaders/HDR/geometry.vert</vertex-shader>
-          <fragment-shader>Shaders/HDR/geometry.frag</fragment-shader>
-          <fragment-shader>Shaders/HDR/gbuffer-include.frag</fragment-shader>
-        </program>
-        <uniform>
-          <name>color_tex</name>
-          <type>sampler-2d</type>
-          <value type="int">0</value>
-        </uniform>
-        <uniform>
-          <name>color_mode</name>
-          <type>int</type>
-          <value><use>material/color-mode</use></value>
-        </uniform>
-        <uniform>
-          <name>material_diffuse</name>
-          <type>float-vec4</type>
-          <value><use>material/diffuse</use></value>
-        </uniform>
-      </pass>
-    </technique>
-
-    <technique n="119">
-      <scheme>hdr-shadow</scheme>
-      <pass>
-        <color-mask type="vec4d">0 0 0 0</color-mask>
-        <cull-face>back</cull-face>
-        <blend>0</blend>
-        <polygon-offset>
-          <factor>1.1</factor>
-          <units>4.0</units>
-        </polygon-offset>
-        <program>
-          <vertex-shader>Shaders/HDR/geometry-shadow.vert</vertex-shader>
-          <fragment-shader>Shaders/HDR/geometry-shadow.frag</fragment-shader>
-        </program>
-      </pass>
-    </technique>
-
+	<!-- HDR removed here -->
 </PropertyList>
