--- a/default.vert
+++ b/default.vert
@@ -1,7 +1,10 @@
 // -*-C++-*-
 
+// CI effect
+// This is FG 2020.4 Shaders/default.vert with a call to CI shader plugged in.
+
 // Shader that uses OpenGL state values to do per-pixel lighting
 //
 // The only light used is gl_LightSource[0], which is assumed to be
 // directional.
 //
@@ -31,10 +34,14 @@
 //uniform int fogType;
 //
 //void fog_Func(int type);
 /////////////////////////
 
+//// CI code include
+void CI_screen_prepare();
+////
+
 void main()
 {
     gl_Position = ftransform();
     ecPosition = gl_ModelViewMatrix * gl_Vertex;
     gl_TexCoord[0] = gl_TextureMatrix[0] * gl_MultiTexCoord0;
@@ -65,6 +72,9 @@
     gl_FrontColor.rgb = constant_term.rgb;  gl_FrontColor.a = 1.0;
     gl_BackColor.rgb = constant_term.rgb; gl_BackColor.a = 0.0;
     //fogCoord = abs(ecPosition.z / ecPosition.w);
 		//fog_Func(fogType);
     setupShadows(ecPosition);
+
+    // CI code insert (this should be the only difference with the standard shader)
+    CI_screen_prepare();
 }
